src/redux/api/quotationApi.tsx

```diff
@@ -0,0 +1,21 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+import { environment } from "@/constants/environment";
+import { IQuotation } from "@/interfaces/quotation";
+
+export const quotationApi = createApi({
+  reducerPath: "quotation",
+  baseQuery: fetchBaseQuery({
+    baseUrl: environment.apiBaseUrl,
+  }),
+  endpoints: (builder) => ({
+    getAllQuotation: builder.query<
+      { data: IQuotation[]; total: number },
+      { offset: number; limit: number; optionalQuery: string }
```

Comment:
optionalQuery   in api 
`&user_id=${searchValue}&sort=${sorting[0]?.id}&desc=${sorting[0]?.desc ? -1: 1}`  
and handle search which uses lodash
is this search one required user ID's data directly from the backend or for other purposes?
not quite understand, can you explain when you have time, thx